File: app/src/main/java/br/com/fernandodutra/testeandroidv2/activities/login/LoginConfigurator.java
package br.com.fernandodutra.testeandroidv2.activities.login;

import java.lang.ref.WeakReference;

/**
 * Created by Fernando Dutra
 * User: Fernando Dutra
 * Date: 26/06/2019
 * Time: 21:16
 * TesteAndroidv2_CleanCode
 */
public enum LoginConfigurator {
    INSTANCE;
    public void configure(LoginActivity activity) {

        LoginRouter router = new LoginRouter();
        router.activity = new WeakReference<>(activity);

        LoginPresenter presenter = new LoginPresenter();
        presenter.loginActivityInputWeakReference = new WeakReference<LoginActivityInput>(activity);

        LoginInteractor interactor = new LoginInteractor();
        interactor.loginPresenterInput = presenter;

        if (activity.loginInteractorInput == null) {
            activity.loginInteractorInput = interactor;
        }
        if (activity.loginRouter == null) {
            activity.loginRouter = router;
        }
    }
}

File: app/src/main/java/com/accenture/primo/bankapp/ui/login/LoginConfigurator.kt
package com.accenture.primo.bankapp.ui.login

enum class LoginConfigurator {
    INSTANCE;

    fun configure(activity: LoginActivity) {
        val router = LoginRouter(activity)
        val presenter = LoginPresenter(activity)
        val interactor = LoginInteractor(presenter)

        if (activity.interactor == null)
            activity.interactor = interactor

        if (activity.router == null)
            activity.router = router
    }
}
